### ISSUE TITLE
Add Groups

### ISSUE THREAD

**Issue by maximihajlov** [merpw/forum#103]
This issue depends on #95, as groups should have invitations.

Events will be handled by another issue.

## Backend

A new table `groups` should be implemented. It will have columns: `primary key`, `title`, `description`  and `creator_id` (foreign key to `users`) and `timestamp`. 

A new table `group_members` should be implemented. It is very similar to `followers` (#94). Columns: `primary key`, `group_id` (foreign key to `groups`), `user_id` (primary key to `users`) and `timestamp`. 

Groups can have posts in it, so `posts` table should be updated with a new optional column `group_id` (foreign key to table `groups`). 

For posts in groups we'll save to the database both author `user_id` and `group_id`). For non-group posts we'll save `null` to `group_id` column. A new `groupId` field will be added to `/posts/create`. 

Several new endpoints should be implemented. It's similar to ones we have for users:

- `GET /api/groups/{id}` - get data about a specific group. It should have the following structure:

```ts
{
    id: 1,
    title: "Amazing group",
    description: "A great place for great posts",

    // similar to `follow_status`, can be `0`-not member, `1`-member and `2`-requested membership
    member_status: 0, 

    members_count: 122, 
}
```

- `GET /api/groups/{id}/posts` - get array of ids of posts of the specific group (sorted by date, newest first). 

- `GET /api/groups` - returns array of ids of all groups, sorted by member count.

- `POST /api/groups/{id}/join` - request joining the group, no request body, available to anyone who is singed up. 

- `POST /api/groups/{id}/invite` - invite user by id, should be available to logged in users who are members of the group. 

```ts
{
    user_id: 1, // id of the user to invite
}
```

- `POST /api/groups/{id}/leave` - leave the group, empty body

Chats and events will be handled in another issue

**Comment by bomanviktor** [comment]
I will finish image upload before lunch. But as you said @, I will not write tests for this since it's a separate service. After that I will get to work woth this!

**Comment by maximihajlov** [comment]
Oh, I forgot about groups creation!

It should be `/api/groups/create`. The concept is similar to post creation. Request body is something like:


```json
{
    "title": "Amazing group"
    "description":  "A great place for great posts"
    "invite": [1, 2, 3]
}
```
Not sure if we have to include "invite", but it's good to have (otherwise the group creation form will be too small)

The response should be an id of the created group, just like for post creation.
